scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1084,22 +1101,22 @@ def test_magnitude(xp):
 def test_magnitude_single_rotation(xp):
     r = Rotation.from_quat(xp.eye(4))
     result1 = r[0].magnitude()
-    xp_assert_close(result1, xp.pi)
+    xp_assert_close(result1, xp.asarray(xp.pi)[()])
 
     result2 = r[3].magnitude()
-    xp_assert_close(result2, 0.0)
+    xp_assert_close(result2, xp.asarray(0.0)[()])
```

Comment:
This highlights a deficiency in `xp_assert_close`. It makes perfect sense for `desired` to be a scalar.